libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -135,8 +141,9 @@ void DataFlowAnalyzer::operator()(Switch& _switch)
 	{
 		unordered_map<YulString, YulString> storage = m_storage;
 		unordered_map<YulString, YulString> memory = m_memory;
+		map<pair<YulString, YulString>, YulString> keccak = m_keccak;
 		(*this)(_case.body);
-		joinKnowledge(storage, memory);
+		joinKnowledge(storage, memory, keccak);
```

Comment:
This is suprising. It does not preserve the precondition stated in the `joinKnowledge()` docstring:

```
/// This only works if the current state is a direct successor of the older point,
/// i.e. `_otherStorage` and `_otherMemory` cannot have additional changes.
```

The state after each case is applied on top of the state after the previous case passed through join, not on top of the initial state from before the `switch`. Which I guess is fine for keccak seeing how `joinKnowledge()` is implemented now  but not sure about memory and storage (I haven't checked how the join helper works for them). Maybe the docstring should be updated to say that the state returned from `joinKnowledge()` is guaranteed to be reusable that way?